File: uce-trade-backend/ms4-search-discovery/internal/core/domain/venture.go
package domain

type Owner struct {
	ID          string `json:"id"`
	FullName    string `json:"fullName"`
	Faculty     string `json:"faculty"`
	PhoneNumber string `json:"phoneNumber"`
	Email       string `json:"email"`
}

type Venture struct {
	ID          string  `json:"id"`
	StudentId   string  `json:"studentId"`
	Title       string  `json:"title"`
	Description string  `json:"description"`
	Category    string  `json:"category"`
	Price       float64 `json:"price"`
	ImageUrl    string  `json:"imageUrl"`
	Status      string  `json:"status"`
	Owner       *Owner  `json:"owner,omitempty"`
}

File: uce-trade-backend/ms2-product-command/src/main/java/ec/edu/uce/trade/product/domain/model/Venture.java
package ec.edu.uce.trade.product.domain.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public class Venture {
    private UUID id;
    private String studentId; 
    private String title;
    private String category;
    private String description;
    private BigDecimal price;
    private String status; 
    private LocalDateTime createdAt;
    private String imageUrl; 

    public Venture() {}

    // Getters y Setters
    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }

    public String getStudentId() { return studentId; }
    public void setStudentId(String studentId) { this.studentId = studentId; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public BigDecimal getPrice() { return price; }
    public void setPrice(BigDecimal price) { this.price = price; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }    
}